sequence_processing_pipeline/Pipeline.py

```diff
@@ -276,6 +297,55 @@ def _validate_sample_sheet(self, sample_sheet_path):
                               isinstance(x, WarningMessage)]
             return val_sheet
 
+    def _validate_mapping_file(self, mapping_file_path):
+        """
+        Performs validation for mapping-files.
+        :return: If successful, an empty list of strings and a valid
+                 mapping-file. If unsuccessful, a list of warning and error
+                 messages and None.
+        """
+        try:
+            df = pd.read_csv(mapping_file_path, delimiter='\t', dtype=str)
+        except pd.errors.ParserError:
+            # ignore parser errors as they obviously prove this is not a
+            # valid mapping file. Return None w/error message
+            raise PipelineError('Cannot parse mapping-file.')
+
+        dupes = Pipeline._identify_duplicate_columns(df.columns)
+
+        if dupes:
+            # column-names are case-insensitive, and must be unique.
+            # return groups of duplicate column names (differentiated only by
+            # a different mixed-case) to the user.
+            raise PipelineError("Mapping-file contains duplicate columns: "
+                                "%s" % ', '.join([str(tpl) for tpl in dupes]))
+
+        obs = {column.lower() for column in df.columns}
```

Comment:
This and the next line can be: 
`df.columns.str.lower()`

Reply:
Thanks! The next line is actually a set, not a dataframe, but I can apply it to the top line.